waves/src/App.tsx

```diff
@@ -14,39 +14,94 @@ export enum AssetType {
     USDT = "USDT",
 }
 
-function App() {
-    const [rate, setRate] = React.useState(191337);
+export type AssetSide = "Alpha" | "Beta";
 
-    const [alphaAsset, setAlphaAsset] = React.useState(AssetType.BTC);
-    const [alphaAmount, setAlphaAmount] = React.useState(0.01);
+export type UpdateAssetAction = { type: "AlphaAmount"; value: number } | { type: "BetaAmount"; value: number } | {
+    type: "AlphaAssetType";
+    value: AssetType;
+} | { type: "BetaAssetType"; value: AssetType };
 
-    const [betaAsset, setBetaAsset] = React.useState(AssetType.USDT);
-    const [betaAmount, setBetaAmount] = React.useState(191.13);
+interface CurrentPairState {
+    alpha: AssetState;
+    beta: AssetState;
+    rate: number;
+}
 
-    const [walletUnlocked, setWalletUnlocked] = React.useState(false);
-    const [publishedTx, setPublishedTx] = React.useState("");
-    const [txPending, setTxPending] = React.useState(false);
+interface AssetState {
+    type: AssetType;
+    amount: number;
+}
 
-    const onUpdateAlphaAssetType = (newType: AssetType) => {
-        console.log(`Received new alpha assetType: ${newType}`);
-        setAlphaAsset(newType);
-    };
-    const onUpdateBetaAssetType = (newType: AssetType) => {
-        console.log(`Received new beta assetType: ${newType}`);
-        setBetaAsset(newType);
-    };
+function reducer(state: CurrentPairState, action: UpdateAssetAction) {
+    switch (action.type) {
+        case "BetaAmount":
```

Comment:
Usually, these actions have a verb in their name to make it clear what they do to the state.